base/compiler/ssair/inlining.jl

```diff
@@ -701,48 +700,57 @@ function resolve_todo(todo::InliningTodo, et::Union{EdgeTracker, Nothing}, cache
                 isconst, src = false, inferred_src
             elseif isa(inferred_src, Const)
                 if !is_inlineable_constant(inferred_src.val)
-                    return compileable_specialization(et, spec.match)
+                    return compileable_specialization(state.et, spec.match)
                 end
                 isconst, src = true, quoted(inferred_src.val)
             end
         end
     else
-        isconst, src = find_inferred(todo.mi, spec.atypes, caches, spec.stmttype)
+        linfo = get(state.mi_cache, todo.mi, nothing)
+        if linfo isa CodeInstance
+            if invoke_api(linfo) == 2
+                # in this case function can be inlined to a constant
+                isconst, src = true, quoted(linfo.rettype_const)
+            else
+                isconst, src = false, linfo.inferred
+            end
+        else
+            isconst, src = false, linfo
+        end
     end
 
-    if isconst && et !== nothing
-        push!(et, todo.mi)
+    if isconst && state.et !== nothing
+        push!(state.et, todo.mi)
         return ConstantCase(src)
     end
 
+    if src !== nothing && !state.policy(src)
+        src = nothing
+    end
+
     if src === nothing
-        return compileable_specialization(et, spec.match)
+        return compileable_specialization(state.et, spec.match)
     end
 
     if isa(src, CodeInfo) || isa(src, Vector{UInt8})
-        src_inferred = ccall(:jl_ir_flag_inferred, Bool, (Any,), src)
-        src_inlineable = ccall(:jl_ir_flag_inlineable, Bool, (Any,), src)
 
```

Comment:
awkwardly blank line? this seems dead?

Reply:
Yeah, you're right. I have some more changes here queued up, so I'll clean this up in a follow up.